src/reporting/clients/MappingsClient.ts

```diff
@@ -0,0 +1,641 @@
+        };
+      }));
+  }
+
+  public async getMapping(accessToken: AccessToken, iModelId: string, mappingId: string): Promise<Mapping> {
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings/${encodeURIComponent(mappingId)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return (await this.fetchJSON<MappingSingle>(url, requestOptions)).mapping;
+  }
+
+  public async createMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mapping: MappingCreate
+  ): Promise<Mapping> {
+    if (!this.isSimpleIdentifier(mapping.mappingName)) {
+      throw new RequiredError(
+        "mappingName",
+        "Required field mappingName was missing or invalid.",
+      );
+    }
+
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings`;
+    const requestOptions: RequestInit = this.createRequest("POST", accessToken, JSON.stringify(mapping));
+    return (await this.fetchJSON<MappingSingle>(url, requestOptions)).mapping;
+  }
+
+  public async updateMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mapping: MappingUpdate
+  ): Promise<Mapping> {
+    if (mapping.description == null && mapping.extractionEnabled == null && mapping.mappingName == null) {
+      throw new RequiredError(
+        "mapping",
+        "All properties of mapping were missing.",
+      );
+    }
+    if (mapping.mappingName != null && !this.isSimpleIdentifier(mapping.mappingName)) {
+      throw new RequiredError(
+        "mappingName",
+        "Required field mappingName was invalid.",
+      );
+    }
+
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings/${encodeURIComponent(mappingId)}`;
+    const requestOptions: RequestInit = this.createRequest("PATCH", accessToken, JSON.stringify(mapping));
+    return (await this.fetchJSON<MappingSingle>(url, requestOptions)).mapping;
+  }
+
+  public async deleteMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string
+  ): Promise<Response> {
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings/${encodeURIComponent(mappingId)}`;
+    const requestOptions: RequestInit = this.createRequest("DELETE", accessToken);
+    return this.fetchJSON<Response>(url, requestOptions);
+  }
+
+  public async copyMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mappingCopy: MappingCopy
+  ): Promise<Mapping> {
+    if (null != mappingCopy.mappingName && !this.isSimpleIdentifier(mappingCopy.mappingName)) {
+      throw new RequiredError(
+        "mappingName",
+        "Field mappingName was invalid.",
+      );
+    }
+    if(this.isNullOrWhitespace(mappingCopy.targetIModelId)) {
+      throw new RequiredError(
+        "targetiModelId",
+        "Required field targetiModelId was missing.",
```

Comment:
`targetIModelId` - the `I` should be uppercase here to match the json property name.